commands/agentstatus.go

```
@@ -120,6 +120,9 @@ func makeFlatMachine(st *state.State, m *state.Machine) (FlatMachine, error) {
 		ID:      m.Id(),
 		Address: address,
 	}
+	if instanceId, err := m.InstanceId(); err == nil {
```

Why ignore the error here? (Or in the bit below?)

---

Laziness. The tools one wasn't me. I've added checks for NotProvisioned and NotFound, respectively, and surfaced the other errors.